File: examples/code/home/use-tools/schedule-tool-execution/schedule_tool.py
from arcade import Arcade
from datetime import datetime, timedelta, timezone

# Initialize the Arcade client
arcade = Arcade()

# Schedule a tool execution for 1 hour in the future
future_time = datetime.now(timezone.utc) + timedelta(hours=1)

# Schedule the GitHub.CreateIssue tool to run at the specified time
scheduled_execution = arcade.tools.schedule(
    tool_name="Github.CreateIssue",
    input={
        "owner": "ArcadeAI",
        "repo": "Docs",
        "title": "Scheduled issue creation",
        "body": "This issue was created by a scheduled tool execution"
    },
    run_at=future_time,
    user_id="user@example.com"  # Optional: specify the user ID for attribution
)

print(f"Scheduled execution ID: {scheduled_execution.id}")
print(f"Status: {scheduled_execution.status}")
print(f"Will run at: {scheduled_execution.run_at}")

# List all scheduled tool executions
scheduled_tools = arcade.tools.list_scheduled()
print(f"Total scheduled executions: {len(scheduled_tools.items)}")

# Get details of a specific scheduled execution
execution_details = arcade.tools.get_scheduled(scheduled_execution.id)
print(f"Tool: {execution_details.tool_name}")
print(f"Status: {execution_details.execution_status}")

# After execution, you can check the attempts and output
if execution_details.attempts:
    latest_attempt = execution_details.attempts[0]
    print(f"Success: {latest_attempt.success}")
    if latest_attempt.success:
        print(f"Output: {latest_attempt.output}")

File: examples/code/home/use-tools/schedule-tool-execution/schedule_tool.js
import { Arcade } from '@arcade-ai/arcade';

// Initialize the Arcade client
const arcade = new Arcade();

// Schedule a tool execution for 1 hour in the future
const futureTime = new Date();
futureTime.setHours(futureTime.getHours() + 1);

// Schedule the GitHub.CreateIssue tool to run at the specified time
async function scheduleToolExecution() {
  try {
    const scheduledExecution = await arcade.tools.schedule({
      toolName: 'Github.CreateIssue',
      input: {
        owner: 'ArcadeAI',
        repo: 'Docs',
        title: 'Scheduled issue creation',
        body: 'This issue was created by a scheduled tool execution'
      },
      runAt: futureTime,
      userId: 'user@example.com' // Optional: specify the user ID for attribution
    });

    console.log(`Scheduled execution ID: ${scheduledExecution.id}`);
    console.log(`Status: ${scheduledExecution.status}`);
    console.log(`Will run at: ${scheduledExecution.runAt}`);

    // List all scheduled tool executions
    const scheduledTools = await arcade.tools.listScheduled();
    console.log(`Total scheduled executions: ${scheduledTools.items.length}`);

    // Get details of a specific scheduled execution
    const executionDetails = await arcade.tools.getScheduled(scheduledExecution.id);
    console.log(`Tool: ${executionDetails.toolName}`);
    console.log(`Status: ${executionDetails.executionStatus}`);

    // After execution, you can check the attempts and output
    if (executionDetails.attempts && executionDetails.attempts.length > 0) {
      const latestAttempt = executionDetails.attempts[0];
      console.log(`Success: ${latestAttempt.success}`);
      if (latestAttempt.success) {
        console.log(`Output: ${JSON.stringify(latestAttempt.output)}`);
      }
    }
  } catch (error) {
    console.error('Error scheduling tool execution:', error);
  }
}

scheduleToolExecution();